src/client.js

```diff
@@ -195,30 +194,16 @@ export class StreamChat {
 			throw new Error('The "id" field on the user is missing');
 		}
 
-		this.userToken = userToken;
-
-		if (userToken == null && this.secret != null) {
-			this.userToken = this.createToken(this.userID);
-		}
-
-		if (this.userToken == null) {
-			throw new Error('both userToken and api secret are not provided');
-		}
-
-		const tokenUserId = UserFromToken(this.userToken);
-		if (
-			userToken != null &&
-			(tokenUserId == null || tokenUserId === '' || tokenUserId !== user.id)
-		) {
-			throw new Error(
-				'userToken does not have a user_id or is not matching with user.id',
-			);
-		}
+		await this._setToken(user, userTokenOrProvider);
 		this._setUser(user);
 		this.anonymous = false;
 
 		return this._setupConnection();
-	}
+	};
+
+	_setToken = async (user, userTokenOrProvider) => {
```

Comment:
ideally we add some validation for `userTokenOrProvider` just to make sure that you dont pass silly stuff to it

Reply:
isn't pointless to have this as an async fn just to await on `this.tokenManager.setTokenOrProvider`

Reply:
> ideally we add some validation for `userTokenOrProvider` just to make sure that you dont pass silly stuff to it

We have validation in token_manager actually